hoptimator-k8s/src/main/java/com/linkedin/hoptimator/k8s/K8sContext.java

```diff
@@ -105,27 +125,29 @@ public static void useContext(K8sContext context) {
     currentContext = context;
   }
 
-  static K8sContext defaultContext() throws IOException {
+
+  // If $HOME/.kube/config is defined, use that config file.
+  // If POD_NAMESPACE_FILEPATH is defined, use that config file.
+  // Use Config.defaultClient() and defaultNamespace if no config file is found.
+  static K8sContext defaultContext(String defaultNamespace) throws IOException {
```

Comment:
Hmm, I think the default namespace is fine to keep "default". I don't see much utility in making this configurable at runtime. It should be the _current_ namespace, or "default" failing that. I don't think we need any further flexibility here.